staticaddr/manager.go

```diff
@@ -0,0 +1,165 @@
+package staticaddr
+
+import (
+	"context"
+	"sync"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/btcec/v2/schnorr"
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/lightninglabs/lndclient"
+	"github.com/lightninglabs/loop"
+	"github.com/lightninglabs/loop/staticaddr/script"
+	"github.com/lightninglabs/loop/swap"
+	staticaddressrpc "github.com/lightninglabs/loop/swapserverrpc"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/keychain"
+)
+
+// ManagerConfig holds the configuration for the address manager.
+type ManagerConfig struct {
+	// AddressClient is the client that communicates with the loop server
+	// to manage static addresses.
+	AddressClient staticaddressrpc.StaticAddressServerClient
+
+	// LoopClient provides loop rpc functionality.
+	SwapClient *loop.Client
+
+	// Store is the database store that is used to store static address
+	// related records.
+	Store AddressStore
+
+	// WalletKit is the wallet client that is used to derive new keys from
+	// lnd's wallet.
+	WalletKit lndclient.WalletKitClient
+
+	// ChainParams is the chain configuration(mainnet, testnet...) this
+	// manager uses.
+	ChainParams *chaincfg.Params
+}
+
+// AddressManager manages the address state machines.
+type AddressManager struct {
```

Comment:
nit: either we should call this `Manager` or the config `AddressManagerConfig` for consistency.